learning/dashboard/angular_routing/Routing/README.md

```diff
@@ -0,0 +1,122 @@
+router.zip file with the reference link. 
+ 
+ 
+ **How to Use Angular Routes**
+
+
+Using Angular routes in a single-page application
+It describes how to build a single-page application, SPA that uses multiple Angular routes.
+
+In a Single Page Application (SPA), all of your application's functions exist in a single HTML page. As users access your application's features, the browser needs to render only the parts that matter to the user, instead of loading a new page. This pattern can significantly improve your application's user experience.
+
+To define how users navigate through your application, you use routes. Add routes to define how users navigate from one part of your application to another. You can also configure routes to guard against unexpected or unauthorized behavior.
+
+
+# DATA PASSING 
+
+
+**Data passing in Angular**
+
+
+In an Angular application, we may have multiple components with a variety of functionality/features and while developing an application we may come across a situation where we need to share or pass the data from one component to another one, in that case, we can achieve that by using the concept of data sharing between the components, and for that in Angular, there are some provisions or ways to achieve the same.
+
+Methods:
+
+  Parent to Child: via Input,
+
+  Child to Parent: via Output() and EventEmitter,
+
+  Child to Parent: via ViewChild,
+
+  Unrelated Components: via a Service.
+
+
+**Why is data passed between Angular components:**
+
+
+Angular provides component based architecture that allows modularizing the application. It means you can create multiple chunks, and convert your large component to a smaller segment that can be easily maintained. The main advantage to do this is easily understanding the component and maintenance of code. In this scenario, it is important for communication between components or share the data between the component. 
+
+Angular provides multiple ways to share the data between the components. Based on your need and kind of component relation, you can select any of the methods. It is recommended to use the service method as it is independent of component relation and helps you to send data to multiple components. 
+
+
+**How to pass the data between angular  routes?**
+
+
+The task is to share data variables between two or more controllers by using AngularJS. There are many procedures to achieve this. Here we will discuss the most popular ones. 
+
+Approach: 
+
+To share data between the controllers in AngularJS we have two main cases:
+
+Share data between parent and child: Here, the sharing of data can be done simply by using controller inheritance as the scope of a child controller inherits from the scope of the parent controller.
+Share data between controllers without having a relation: Here, the sharing of data can be done in a few ways some of them are: 
+
+By using the rootScope variable:
```

Comment:
This is from AngularJS, which is the very first version of Angular.
Currently this is not followed.
Update the Approach section based on Angular 2+ (current version is 15. You can follow something that is based on Angular 14)